opm/input/eclipse/Schedule/ScheduleState.hpp

```diff
@@ -500,6 +500,9 @@ namespace Opm {
         // constant flux aquifers
         std::unordered_map<int, SingleAquiferFlux> aqufluxs;
         BCProp bcprop;
+        // injection streams for compostional STREAM injection using WINJGAS 
+        std::unordered_map<std::string, std::vector<double>> inj_streams;
```

Comment:
Please use `map_member<>`.  It really is the primary solution to this problem.

Reply:
Sure.  I went back and forth on this  and thanks for the suggestion.  I will do that. 

not very familiar with the design here.  Will you explain a little bit what is the difference here for future record?

Reply:
> not very familiar with the design here. Will you explain a little bit what is the difference here for future record?

Sure.  The point is that `map_member<K,V>` is more or less a `map<K, shared_ptr<V>>` with some additional logic on top.  This enables sharing data between `ScheduleState` objects at different report steps if there is no intervening keyword to add/change the data.

Reply:
The design looks like `V` needs to be a struct containing `name()` , which is the key .

Reply:
Is it okay that I add a member function 

```c++
void update(const K& key, std::shared_ptr<T> value);
```

or you think it will break the design.

Reply:
It looks it might work, but it will be on draft mode for some while until we figure out how things work in the opm-simulators.  I will check what Jenkins think about it. 

If it does not later, I will make a StreamComposition struct to make it the same with Well or Group to following the style change.